### PR TITLE
 Implement validator index reports table

### LINKED ISSUE #807: Implement validator index reports table

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed #807

### CODE REVIEW

File: scan/src/components/validator/ReportsTable.re [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                </TBody>
+              })
+            ->Array.of_list
+            ->React.array}
+         </>
+       : <div className=Styles.iconWrapper>
+           <VSpacing size={`px(30)} />
+           <img src=Images.noRequestIcon className=Styles.icon />
+           <VSpacing size={`px(40)} />
+           <Text block=true value="NO BLOCK" weight=Text.Regular color=Colors.blue4 />
```
Comment (taobun) [inline scan/src/components/validator/ReportsTable.re]: NO REPORTS

File: scan/src/components/validator/ReportsTable.re [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+           <VSpacing size={`px(15)} />
+         </div>}
```
Comment (taobun) [inline scan/src/components/validator/ReportsTable.re:238]: Add pagination and mock pageCount

File: scan/src/components/validator/ReportsTable.re [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+  let reports = [
+    (
+      324,
+      "6F45B0D19B46F144CDD7ACA9674E2AD8E8F8C15EF56CA073749B2ACD7DF7739D",
+      234,
+      "Mean Crypto Price",
+      [23, 12, 35],
+      [1, 2, 3],
+      [123213123, 123123132, 123123123],
+    ),
```
Comment (taobun) [inline scan/src/components/validator/ReportsTable.re:39]: Add 1 more report, I want to see table more

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/components/validator/ReportsTable.re [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+                ) => {
+                <TBody key=hash>
+                  <Row>
+                    <Col> <HSpacing size=Spacing.md /> </Col>
+                    <Col size=1. alignSelf=Col.Start>
+                      <TypeID.Request id={ID.Request.ID(requestID)} />
+                    </Col>
+                    <Col size=2. alignSelf=Col.Start>
+                      <div className={Styles.withWidth(140)}>
+                        <Text value=hash block=true code=true ellipsis=true />
```
Comment (taobun) [inline scan/src/components/validator/ReportsTable.re:162]: set font weight Medium

**Review by taobun — CHANGES_REQUESTED** [review]

**Comment by taobun** [comment]
@Benzbeeb Fix conflict first

**Review by prin-r — DISMISSED** [review]

File: scan/src/pages/ValidatorIndexPage.re [hunk 5]
```diff
@@ -166,8 +166,8 @@ let make = (~address, ~hashtag: Route.validator_tab_t) => {
       currentRoute={Route.ValidatorIndexPage(address, hashtag)}>
       {switch (hashtag) {
        | ProposedBlocks => <ProposedBlocksTable />
-       | Delegators => <div> <DelegatorsTable /> </div>
-       | Reports => <div> <Text value="Reports" /> </div>
+       | Delegators => <div> <Text value="Delegators" /> </div>
```
Comment (taobun) [inline scan/src/pages/ValidatorIndexPage.re]: LUL

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/pages/ValidatorIndexPage.re (hunk 5 above)
Comment (Benzbeeb) [reply scan/src/pages/ValidatorIndexPage.re]: Shame on me

File: scan/src/components/validator/ReportsTable.re [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+                       ->Belt.List.toArray
+                       ->React.array}
+                    </Col>
+                    <Col> <HSpacing size=Spacing.lg /> </Col>
+                  </Row>
+                </TBody>
+              })
+            ->Array.of_list
+            ->React.array}
+           //  <Pagination currentPage="1" pageCount onPageChange={newPage => setPage(_ => newPage)} />
```
Comment (taobun) [inline scan/src/components/validator/ReportsTable.re]: LUL

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/components/validator/ReportsTable.re [hunk 7]
```diff
@@ -0,0 +1,243 @@
+module Styles = {
+  open Css;
+
+  let vFlex = align => style([display(`flex), flexDirection(`row), alignItems(align)]);
+
+  let tableWrapper = style([padding2(~v=`px(9), ~h=`px(15))]);
+
+  let fixWidth = style([width(`px(230))]);
```
Comment (evilpeach) [inline scan/src/components/validator/ReportsTable.re]: remove

File: scan/src/components/validator/ReportsTable.re [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+                      <div className={Styles.withWidth(140)}>
+                        <Text value=hash block=true code=true ellipsis=true />
+                      </div>
+                    </Col>
+                    <Col size=2. alignSelf=Col.Start>
+                      <Row>
+                        <TypeID.OracleScript id={ID.OracleScript.ID(oracleScriptID)} />
+                        <HSpacing size=Spacing.sm />
+                        <HSpacing size=Spacing.xs />
+                        <Text value=oracleScriptDescription block=true code=true />
```
Comment (evilpeach) [inline scan/src/components/validator/ReportsTable.re]: support `text-overflow` if the length of the word exceeds max-width.

File: scan/src/components/validator/ReportsTable.re [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+                      </Row>
+                    </Col>
+                    <Col size=1.>
```
Comment (evilpeach) [inline scan/src/components/validator/ReportsTable.re:175]: Add more space between OracleScript column and DataSource column

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by prin-r — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]
LGTM Thanks!

**Pull request merged**